src/transformStateWithClones.js

```diff
@@ -1,13 +1,39 @@
 'use strict';
-
+/* eslint-disable */
 /**
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const historyChanges = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newState = {
+          ...newState,
+          ...action.extraData,
+        };
+        break;
+      case 'removeProperties':
+        for (const fieldName of action.keysToRemove) {
+          delete newState[fieldName];
+        }
+        break;
+      case 'clear':
+        newState = {};
+        break;
+      default:
+        break;
```

Comment:
`default`: present, but it does not perform any actions. You can add a throw to handle errors.
Example: 
```suggestion

      default:
        throw new Error(`Unknown action type: ${action.type}`);
```